commands/sketch/new.go

```diff
@@ -55,19 +55,32 @@ func NewSketch(ctx context.Context, req *rpc.NewSketchRequest) (*rpc.NewSketchRe
 		return nil, err
 	}
 
+	templateDir := configuration.Settings.GetString("directories.template")
 	sketchDirPath := paths.New(sketchesDir).Join(req.SketchName)
-	if err := sketchDirPath.MkdirAll(); err != nil {
-		return nil, &arduino.CantCreateSketchError{Cause: err}
-	}
-	sketchName := sketchDirPath.Base()
-	sketchMainFilePath := sketchDirPath.Join(sketchName + globals.MainFileValidExtension)
-	if !req.Overwrite {
-		if sketchMainFilePath.Exist() {
-			return nil, &arduino.CantCreateSketchError{Cause: errors.New(tr(".ino file already exists"))}
+
+	var sketchMainFilePath *paths.Path
+	if templateDir != "" {
+		templateDirPath := paths.New(templateDir)
+		if err := templateDirPath.CopyDirTo(sketchDirPath); err != nil {
+			return nil, &arduino.CantCreateSketchError{Cause: err}
+		}
+		// TODO: Make this customizable?
+		sketchMainFilePath = sketchDirPath.Join("main.ino")
```

Comment:
This won't work, `main.ino` must be named the same as the sketch folder (`/path/to/my_sketch/my_sketch.ino` for example).
Since the sketch folder is given as a command line argument by the user, you should do something similar to the `else` side of the `if`:
```go
sketchName := sketchDirPath.Base()
sketchMainFilePath = sketchDirPath.Join(sketchName + globals.MainFileValidExtension)
```
probably these two lines can be factored out of the if.